### PR TITLE
Add 304 release notes

### PR DESCRIPTION
**Pull request by martint**

### CODE REVIEW

File: presto-docs/src/main/sphinx/release/release-304.rst [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+Release 304
+===========
+
+General Changes
+---------------
+
+* Fix wrong results for queries involving ``FULL OUTER JOIN`` and ``coalesce`` expressions
+  over the join keys. (:issue:`288`)
+* Fix failure when column from view is referenced using its fully qualified form. (:issue:`250`)
+* Add experimental support for Spill to Disk for OrderBy.
```
Comment (sopel39) [inline presto-docs/src/main/sphinx/release/release-304.rst]: (:issue:`228`)

File: presto-docs/src/main/sphinx/release/release-304.rst [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+* Add experimental support for Spill to Disk for WindowOperator.
+* Correctly report internal network/physical position count for operators.
```
Comment (sopel39) [inline presto-docs/src/main/sphinx/release/release-304.rst]: (:issue:`271`)

File: presto-docs/src/main/sphinx/release/release-304.rst [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+* Sum internal network/physical data size and position count from all stages in query statistics.
+* Make planning more deterministic.
```
Comment (sopel39) [inline presto-docs/src/main/sphinx/release/release-304.rst]: (:issue:`226`)

File: presto-docs/src/main/sphinx/release/release-304.rst [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+* Remove deprecated ``datasources`` configuration property. (:issue:`306`)
+* Do not remove modified configuration files during RPM uninstallation. (:issue:`304`)
+* Improve error message when a query contains zero-length delimited identifiers. (:issue:`249`)
+
+Web UI Changes
+--------------
+
+* Fix broken timeline view.
```
Comment (sopel39) [inline presto-docs/src/main/sphinx/release/release-304.rst]: (:issue:`283`)

File: presto-docs/src/main/sphinx/release/release-304.rst [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+* Report internal network/physical data size and position count in detailed query view.
```
Comment (sopel39) [inline presto-docs/src/main/sphinx/release/release-304.rst]: (:issue:`271`)

**Comment by findepi** [comment]
I didn't add RNs for
* [e9c217a](https://github.com/prestosql/presto/commit/e9c217ae4a3e1a3c73be5bcd417bb5cf1384eaca) Disable Airlift admin servlet
* [6c4a28d](https://github.com/prestosql/presto/commit/6c4a28dab8608bbf285e8d2bca57fb3ba690798f) Disable Airlift admin servlet in Proxy

however, from user perspective, nothing really changed. IMO we're ok not mentioning this.

**Comment by martint** [comment]
> however, from user perspective, nothing really changed. IMO we're ok not mentioning this.

I think we should. It's something admins might notice or care about.

**Comment by findepi** [comment]
@martint please help me word this. I came up with this:
```
General
---
* Stop opening a random unused listening network port. (issue:`260`)
```
(or should it link to #239 (the issue) instead of #260 (the PR)?

**Comment by martint** [comment]
Thinking out loud... how about: "Avoid opening an unnecessary HTTP listener on a random port"

Also, in general, I'd link to the issue if one's available.

**Comment by electrum** [comment]
Hive:
```
* Support Parquet files written with Parquet 1.9+ that use ``DELTA_BINARY_PACKED``
  encoding with ``INT64``. (:issue:`334`)
```

File: presto-docs/src/main/sphinx/release/release-304.rst [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+  that contain capital letters. (:issue:`289`)
+* Fix wrong results when filtering on the hidden ``$bucket`` column for tables containing
+  partitions with different bucket counts. Instead, queries will now fail in this case. (:issue:`286`)
+* Record the configured Hive time zone when writing ORC files. (:issue:`212`)
+* Use the time zone recorded in ORC files when reading timestamps.
+  The configured Hive time zone, which was previously always used, is now
+  used only as a default when the writer did not record the time zone. (:issue:`212`)
+* Support Parquet files written with Parquet 1.9+ that use ``DELTA_BINARY_PACKED``
+  encoding with ``INT64``. (:issue:`334`)
+* Allow configuring the retry policy for the Thrift metastore client. (:issue:`240`)
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-304.rst]:
```
* Allow configuring the retry policy for the Thrift metastore client using the
  ``hive.metastore.thrift.client.*`` configuration properties. (:issue:`240`)

**Pull request merged**